core/core.go

```diff
@@ -65,7 +69,7 @@ type Driver interface {
 type NetworkDriver interface {
 	// A network driver implements the programming logic for network
 	Driver
-	Init(config *Config, stateDriver StateDriver) error
+	Init(config *Config, info *InstanceInfo, stateDriver StateDriver) error
```

Comment:
@jainvipin 

`Config` and `InstanceInfo` still are separate. InstanceInfo provides a way to pass plugin-instance specific info down to the drivers which are more runtime on nature (like host-label). 

I agree that we should define core API in a way that they change little when a new parameter shows up. Thinking more, I feel we can classify parameters passed to `Init()` into two categories viz. config and instance-specific. That way we can fold the StateDriver argument into `InstanceInfo` as well. Going forward the `InstanceInfo` structure can then provide a way to add mroe params (if any) that may need to be passed to drivers. WDYT? I can make the change if that sounds ok.